### DISCUSSION TITLE
Question about rendering more complex shapes

### DISCUSSION THREAD

**Discussion by Huda-sh** [MuhammadSulaiman001/opengl-lab#31] [Q&A]
Greetings✨,

I am trying to draw a specific shape in the project, but I am finding it difficult to get the vertices and indices of it, so I am if thinking of making it out of simple shapes (cube and sphere).
I have seen that in the theoretical lectures Dr. Ammar talked about Constructive Solid Geometry Methods (CSG), where we can apply the union, intersect, and difference operations on simple shapes to create more complex ones.

![an image referencing the lecture slide where Dr. Ammar mentioned CSG](https://github.com/MuhammadSulaiman001/opengl-lab/assets/68558969/5a132611-9105-4f66-bfda-7195b7f35346)

can you please tell me how to apply that in modern OpenGL, or recommend a resource so I can check it

I came across a discussion on StackOverFlow [here](https://stackoverflow.com/questions/22463634/is-it-possible-to-create-unions-of-primitives-in-opengl) where someone mentioned the [OpenCSG](http://opencsg.org/) library, but I don't know if it is allowed to use it in the project, if it is allowed please tell me.

thank you in advance 🙏🏻

**Comment by MuhammadSulaiman001** [comment]
Hello @Huda-sh,

Even though I've never came across CSG,  I liked the idea. I believe that the native OpenGL API does not support CSG, therefore, I can give you a permission to use any library that implements CSG methods.

Note: I've checked OpenCSG [repository](https://github.com/floriankirsch/OpenCSG) on GitHub, I've read the [example](https://github.com/floriankirsch/OpenCSG/blob/master/example/main.cpp), it uses legacy OpenGL API.. never mind, I want to tell you that you can mix your code with both legacy and modern OpenGL code (but I wonder if using GLUT (or FreeGLUT) will cause any technical issues with your existing code, I hope it will not).

Good luck.
